var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -92,6 +92,9 @@ class Charm(Package):
     variant("production", default=True, description="Build charm++ with all optimizations")
     variant("tracing", default=False, description="Enable tracing modules")
 
+    provides('mpi', when='build-target=AMPI')
+    provides('mpi', when='build-target=LIBS')
+
     depends_on("mpi", when="backend=mpi")
```

Comment:
This is currently causing a circular dependency regarding MPI when building with ```backend=mpi```. Not sure how to fix this.